### ISSUE TITLE
Variable chart standards and quality

### ISSUE THREAD

**Issue by AntonOfTheWoods** [bitnami/charts#4237]
Many of the charts seem to be at various levels of sophistication and standardisation. I was thinking of basically trying to use bitnami everything where something was available, so I have a coherent style and set of patterns (and so I can rely on things like your `common`). I see that there are varying levels of conformity to your own standards though, so wanted to know whether there is an real example of a "model chart" that you guys want to move towards (with more than is in the model template). If we take, for example, `nginx`, `postgresql`, `postgresql-ha` and `minio`, there are either subtle differences (with `prometheus` metrics/`ServiceMonitor`s, for example) or charts that don't appear to have felt much love (`minio`, doesn't use common, doesn't have a `ServiceMonitor`, and what needs to be done to get `prometheus` to scrape it is very unclear). The `minio` `pvc` also doesn't have any possibility to add `annotation`s, so when a standalone deployment is removed, all the data is also deleted, with no way to specify to keep it (like there is with the `harbor` charts `pvc`s with `persistence.resourcePolicy`). The `minio` chart seems to be being updated, so it doesn't appear to have been abandoned. 

I see also that the template still has things like a `requirements.yaml`, which will start being actively removed from existing repositories starting from next week. Is there an ETA on having an up-to-date template that corresponds to where bitnami wants things to move?

Basically, I can try and submit a few PRs if you can tell me which charts I should copy/paste from :-)!

**Comment by carrodher** [comment]
Thanks for creating this issue; you're right, as the Bitnami Helm Charts have been created in different moments the standardizations have been varying. We are working on unifying all the Helm Charts little by little in order to incorporate the best practices and standards.

Recently we created [this scaffolding](https://github.com/bitnami/charts/tree/master/template) in order to serve as a base when creating new Helm Charts, in the same way, there are a couple of blog posts describing the best practices used when developing a Helm Chart, such as https://docs.bitnami.com/tutorials/production-ready-charts/.

Thinking about a recent addition, you can use wavefront as an example since it was one of the recent Helm Charts added to the repository. We will be happy to review any PR unifying the standard.

About the _requirements.yaml_ deprecation, please see this issue (https://github.com/bitnami/charts/issues/3867), we have a roadmap ready to start migrating all the Bitnami Helm Charts to the new `apiVersion` as well as other related changes starting this Friday (November 13th). From that moment, the Bitnami Helm Charts are going to use the Helm 3 format, removing the Helm 2 support.

**Comment by AntonOfTheWoods** [comment]
Thanks for that! I see [keycloak](https://github.com/bitnami/charts/tree/master/bitnami/keycloak) is hot off the press, and actually contains lots of the stuff I am using myself. If you can confirm it's considered a good example of where you want to head, that would be great to get me moving in the right direction. 

I will try and submit some patches once you have done the migration, particularly for charts like `minio`.

**Comment by carrodher** [comment]
Yes, it can be a good example, it is also a recent addition, feel free to use it as the base.
